ui/src/app/applications/components/applications-list/applications-filter.tsx

```diff
@@ -2,58 +2,112 @@ import {useData, Checkbox} from 'argo-ui/v2';
 import * as minimatch from 'minimatch';
 import * as React from 'react';
 import {Context} from '../../../shared/context';
-import {Application, ApplicationDestination, Cluster, HealthStatusCode, HealthStatuses, SyncPolicy, SyncStatusCode, SyncStatuses} from '../../../shared/models';
-import {AppsListPreferences, services} from '../../../shared/services';
+import {
+    AbstractApplication,
+    Application,
+    ApplicationDestination,
+    ApplicationSet,
+    ApplicationSetConditionStatuses,
+    ApplicationSetSpec,
+    ApplicationSetStatus,
+    ApplicationSpec,
+    ApplicationStatus,
+    Cluster,
+    HealthStatusCode,
+    HealthStatuses,
+    Operation,
+    SyncPolicy,
+    SyncStatusCode,
+    SyncStatuses
+} from '../../../shared/models';
+import {AbstractAppsListPreferences, AppSetsListPreferences, AppsListPreferences, services} from '../../../shared/services';
 import {Filter, FiltersGroup} from '../filter/filter';
 import * as LabelSelector from '../label-selector';
-import {ComparisonStatusIcon, getAppDefaultSource, HealthStatusIcon} from '../utils';
+import {ComparisonStatusIcon, getAppDefaultSource, getAppSetHealthStatus, HealthStatusIcon, isApp, isInvokedFromAppsPath} from '../utils';
+import {ContextApis} from '../../../shared/context';
+import {History} from 'history';
 
-export interface FilterResult {
+export interface AbstractFilterResult {
+    favourite: boolean;
+    labels: boolean;
+    health: boolean;
+}
```

Comment:
@reggie-k , double-checking my rebase and is 

`namespaces: boolean;`

missing here?  It got pulled out from the original FilterResult